nixos/modules/services/cluster/dynomite.nix

```diff
@@ -0,0 +1,82 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+let
+  cfg = config.services.dynomite;
+  configFile = pkgs.writeText "dynomite.yaml" "${ cfg.config }";
+in {
+  options = {
+    services.dynomite = {
+      enable = mkEnableOption "dynomite server";
+       config = mkOption {
+        default = "";
+        example = ''
+          dyn_o_mite:
+            listen: 127.0.0.1:8102
+            dyn_listen: 127.0.0.1:8101
+            tokens: '101134286'
+            servers:
+            - 127.0.0.1:6379:1
+            data_store: 0
+            mbuf_size: 16384
+            max_msgs: 300000
+        '';
+        type = types.lines;
+        description = ''
+          Configuration for dynomite, see <link xlink:href="https://github.com/Netflix/dynomite"/>
+        '';
+      };
+      # Enable as soon as dynomite supports YAML 1.2
+      #settings = mkOption {
+      #  type = types.attrs;
+      #  default = {
+      #    dyn_o_mite = {
+      #      listen = "127.0.0.1:8102";
+      #      dyn_listen =  "127.0.0.1:8101";
+      #      tokens =  "'101134286'";
+      #      servers = [ "127.0.0.1:${config.services.redis.port}:1" ];
+      #      data_store = 0;
+      #      mbuf_size = 16384;
+      #      max_msgs = 300000;
+      #    };
+      #  };
+      #  description = ''
+      #    Configuration for dynomite, see <link xlink:href="https://github.com/Netflix/dynomite"/>
+      #  '';
+      #};
+    };
+  };
+
+  config = mkIf cfg.enable {
+
+    # Enable as soon as dynomite supports YAML 1.2
+    #services.redis.enable = any (value: value == "127.0.0.1:${toString config.services.redis.port}:1") cfg.settings.dyn_o_mite.servers;
+
+
+    systemd.services.dynomite = {
+      description = "Dynomite";
+      wantedBy = [ "multi-user.target" ];
+      after = ["network.target" "redis.service" ];
+      requires = [ "redis.service" ];
```

Comment:
Given that `redis` isn't enabled does this make sense? If you unconditionally enable this now it would break compatibility to conditionally enable it later.

Might it be worthwhile to consider a NixOS option for `servers` to make this more intelligent from the start? Never having used `dynomite` I'm not sure what the standard workflow is and how NixOS might provide value to users, but after a quick read of the site I thought a `servers` option seemed reasonable as the application can't function without that value.

Reply:
@Scriptkiddi This doesn't look resolved? Why did you mark it as such?

Reply:
I mixed it up with the service config, will fix it